CRM/MembershipExtras/Hook/Pre/MembershipPaymentPlanProcessor.php

```diff
@@ -113,11 +113,29 @@ private function alterFirstContributionParameters() {
     $this->params['total_amount'] =  $this->recurringContribution['amount'];
     $this->params['net_amount'] =  $this->recurringContribution['amount'];
 
-    if (!empty($this->params['tax_amount'])) {
+    $lineItemCount = $this->countLineITems();
```

Comment:
I don't  think that this is a good solution, because you can have a price set with a single line item, maybe there is some parameters that will be passed (like price set id) that are better suited to detect if this a price set of finical type contribution ?